frame/support/src/inherent.rs

```diff
@@ -44,12 +52,12 @@ pub trait ProvideInherent {
 	/// implementation returns this.
 	///
 	/// - `Ok(Some(e))` indicates that this inherent is required in this block. `construct_runtime!`
-	/// will call this function from in its implementation of `fn check_extrinsics`.
+	/// will call this function in its implementation of `fn check_extrinsics`.
 	/// If the inherent is not present, it will return `e`.
 	///
 	/// - `Err(_)` indicates that this function failed and further operations should be aborted.
 	///
-	/// NOTE: If inherent is required then the runtime asserts that the block contains at least
+	/// NOTE: If the inherent is required then the runtime asserts that the block contains at least
 	/// one inherent for which:
 	/// * type is [`Self::Call`],
 	/// * [`Self::is_inherent`] returns true.
```

Comment:
I'm not sure how to write it but this `is_inherent_required` is currently only checked by other block producer same as `check_inherent`, maybe we can change it to make it part of the `execute_block`. But it is not the case currently.